### PR TITLE
Fix several NixOS errors in Hydra evaluation

### PR DESCRIPTION
**Pull request by rnhmjoj**
###### Motivation for this change

A cleaner hydra-eval-jobs output.

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

You can get a list of mostly relevant errors with

```shell
curl -s https://hydra.nixos.org/jobset/nixos/trunk-combined | \
  tail -n +221 | \
  head -n -212 | \
  sed \
    -e '/insecure, refusing to evaluate./,+32d' \
    -e '/as broken, refusing to evaluate./,+13d' \
    -e '/unfree license.\+refusing to evaluate./,+19d' \
    -e '/is not supported on /,+13d' \
    -e '/Failed to evaluate.\+unfree\|broken\|insecure/d' \
    -e '/warning: error: SQLite database/d' \
    -e '/android-sdk-license/,+143d' \
    -e '/not supported for interpreter/d' \
    >  clean-log
```

### CODE REVIEW

File: nixos/tests/postfixadmin.nix [hunk 1]
```diff
@@ -1,6 +1,6 @@
 import ./make-test-python.nix ({ pkgs, ...} : {
   name = "postfixadmin";
-  meta = with pkgs.stdenv.lib.maintainers; {
```
Comment (SuperSandro2000) [inline nixos/tests/postfixadmin.nix:3]: @siraben Not sure how we missed that.

Comment (rnhmjoj) [reply nixos/tests/postfixadmin.nix:3]: Same, It must have been added after the treewide PRs, but I haven't checked.

Comment (Atemu) [reply nixos/tests/postfixadmin.nix:3]: Perhaps we could add some sort of ofBorg check or similar for stdenv.lib.

File: nixos/modules/programs/atop.nix [hunk 2]
```diff
@@ -141,8 +141,15 @@ in
...
-      security.wrappers =
-        lib.mkIf cfg.setuidWrapper.enable { atop = { source = "${atop}/bin/atop"; }; };
+
+      security.wrappers = lib.mkIf cfg.setuidWrapper.enable {
+        atop =
+          { setuid = true;
+            owner = "root";
+            group = "root";
+            source = "${atop}/bin/atop";
+          };
```
Comment (SuperSandro2000) [inline nixos/modules/programs/atop.nix:151]:
```suggestion
        atop = {
          setuid = true;
          owner = "root";
          group = "root";
          source = "${atop}/bin/atop";
        };
```

**Comment by rnhmjoj** [comment]
After shaving off all unrelated or unavoidable warnings (testing if the warning works), what's left is:
```
# 1. ??
error: attribute 'nixos.tests.installed-tests.recurseForDerivations' is a Boolean, which is not supported
error: attribute 'nixos.tests.kernel-generic.testsForKernel' is a function, which is not supported
error: attribute 'nixos.tests.kernel-generic.testsForLinuxPackages' is a function, which is not supported

# 2. ??
error: value is a function while a set was expected

       at /nix/store/9c5skd55qq227fswpw7h7g6w2srvxzld-source/lib/attrsets.nix:381:46:

          380|   */
          381|   zipAttrsWith = f: sets: zipAttrsWithNames (concatMap attrNames sets) f sets;
             |                                              ^
          382|   /* Like `zipAttrsWith' with `(name: values: values)' as the function.

# 3. nixos/tests/installer.nix
error: Non-EFI boot methods are only supported on i686 / x86_64

# 4. nixos/modules/services/web-servers/trafficserver.nix
error: attempted to realize '/nix/store/cy1bhfwij3labr8lb0d7llwn3kjzpjpx-in.json.drv' during evaluation but 'allow-import-from-derivation' is false
error: attempted to realize '/nix/store/aasc4xnqag4nl4z286v2bc3cf656q85c-in.json.drv' during evaluation but 'allow-import-from-derivation' is false
error: attempted to realize '/nix/store/k7c0c7s1qxb1npwfzhd8d27jllw7iz0a-in.json.drv' during evaluation but 'allow-import-from-derivation' is false
```

1. I'm can't find the source of the error: who's assuming that every output in the test must be a derivation?
2. No idea, way too many references
3. who's trying to build the installer test on non-x86?
4. Is this expected on hydra? I can build `nixos/tests/trafficserver.nix` locally

**Comment by vcunat** [comment]
3.  aarch64-linux, most likely; we have successful ones, e.g. https://hydra.nixos.org/job/nixos/trunk-combined/nixos.tests.installer.simpleUefiGrubSpecialisation.aarch64-linux
4.  Yes, official NixPkgs should be without IFD, at least on normal evaluations.

**Comment by mohe2015** [comment]
For 4. e.g. on nix unstable `nix --option allow-import-from-derivation false build github:NixOS/nixpkgs#nixosTests.trafficserver` fails.

**Review by vcunat — APPROVED** [review]
Looks good to me.  I tried running most of the changed tests, too.

**Comment by vcunat** [comment]
Thanks a lot!

Let's get Hydra moving; we can do further tweaks in subsequent PRs.

**Comment by vcunat** [comment]
Hmm, kubernetes tests have regressed in the past few days (i.e. probably around this PR): https://hydra.nixos.org/eval/1706946#tabs-now-fail

**Comment by rnhmjoj** [comment]
Uhm, looks like it: the only related [changes](https://hydra.nixos.org/api/scmdiff?rev2=a930f7da84786807bb105df40e76b541604c3e72&type=git&branch=&rev1=bc9b956714ed6eac5f8888322aac5bc41389defa&uri=https%3A%2F%2Fgithub.com%2FNixOS%2Fnixpkgs.git) seems to be my commit and a minor kubernetes update.

File: nixos/tests/postfixadmin.nix (hunk 1 above)
Comment (siraben) [reply nixos/tests/postfixadmin.nix:3]: @SuperSandro2000 #138918

**Comment by rnhmjoj** [comment]
4. should be fixed with https://github.com/NixOS/nixpkgs/pull/138964

**Comment by vcunat** [comment]
The [`plotinus` test](https://hydra.nixos.org/job/nixos/trunk-combined/nixos.tests.plotinus.x86_64-linux) doesn't work either, but it didn't evaluate for a year...

**Pull request merged**